src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ReadOnlySpanMarshaller.cs

```diff
@@ -166,5 +167,58 @@ public void Free()
                 return ref MemoryMarshal.GetReference(managed);
             }
         }
+
+        /// <summary>
+        /// Supports marshalling from unmanaged to managed in a call from managed code to unmanaged code. For example, return values and `out` parameters in P/Invoke methods.
+        /// </summary>
+        public struct ManagedToUnmanagedOut
+        {
+            private TUnmanagedElement* _unmanagedArray;
+            private T[]? _managedValues;
+
+            /// <summary>
+            /// Initializes the <see cref="ReadOnlySpanMarshaller{T, TUnmanagedElement}.ManagedToUnmanagedOut"/> marshaller.
+            /// </summary>
+            /// <param name="unmanaged">A pointer to the array to be unmarshalled from native to managed.</param>
+            public void FromUnmanaged(TUnmanagedElement* unmanaged)
+            {
+                _unmanagedArray = unmanaged;
+            }
+
+            /// <summary>
+            /// Returns the managed value representing the native array.
+            /// </summary>
+            public ReadOnlySpan<T> ToManaged()
+            {
+                return new ReadOnlySpan<T>(_managedValues!);
+            }
+
+            /// <summary>
+            /// Returns a span that points to the memory where the unmanaged elements of the array are stored.
+            /// </summary>
+            /// <returns>A span over unmanaged values of the array.</returns>
+            public ReadOnlySpan<TUnmanagedElement> GetUnmanagedValuesSource(int numElements)
+            {
+                return new ReadOnlySpan<TUnmanagedElement> ( _unmanagedArray, numElements);
+            }
+
+            /// <summary>
+            /// Returns a span that points to the memory where the managed elements of the array should be stored.
+            /// </summary>
+            /// <returns>A span where managed values of the array should be stored.</returns>
+            public Span<T> GetManagedValuesDestination(int numElements)
+            {
+                _managedValues = new T[numElements];
+                return _managedValues;
+            }
+
+            /// <summary>
+            /// Frees resources.
+            /// </summary>
+            public void Free()
+            {
+                NativeMemory.Free(_unmanagedArray);
```

Comment:
I think that this should be `Marshal.FreeCoTaskMem((IntPtr)_unmanagedArray);`

@jkoritzinsky Can you please confirm.

Reply:
Yep this should be `FreeCoTaskMem`. Good catch!